votingpool/withdrawal.go

```diff
@@ -0,0 +1,939 @@
+type OutBailmentOutpoint struct {
+	ntxid  Ntxid
+	index  uint32
+	amount btcutil.Amount
+}
+
+// changeAwareTx is just a wrapper around wire.MsgTx that knows about its change
+// output, if any.
+type changeAwareTx struct {
+	*wire.MsgTx
+	changeIdx int32 // -1 if there's no change output.
+}
+
+// WithdrawalStatus contains the details of a processed withdrawal, including
+// the status of each requested output, the total amount of network fees and the
+// next input and change addresses to use in a subsequent withdrawal request.
+type WithdrawalStatus struct {
+	nextInputAddr  WithdrawalAddress
+	nextChangeAddr ChangeAddress
+	fees           btcutil.Amount
+	outputs        map[OutBailmentID]*WithdrawalOutput
+	sigs           map[Ntxid]TxSigs
+	transactions   map[Ntxid]changeAwareTx
+}
+
+// TxSigs is list of raw signatures (one for every pubkey in the multi-sig
+// script) for a given transaction input. They should match the order of pubkeys
+// in the script and an empty RawSig should be used when the private key for a
+// pubkey is not known.
+type TxSigs [][]RawSig
+
+// RawSig represents one of the signatures included in the unlocking script of
+// inputs spending from P2SH UTXOs.
+type RawSig []byte
+
+// byAmount defines the methods needed to satisify sort.Interface to
+// sort a slice of OutputRequests by their amount.
+type byAmount []OutputRequest
+
+func (u byAmount) Len() int           { return len(u) }
+func (u byAmount) Less(i, j int) bool { return u[i].Amount < u[j].Amount }
+func (u byAmount) Swap(i, j int)      { u[i], u[j] = u[j], u[i] }
+
+// byOutBailmentID defines the methods needed to satisify sort.Interface to sort
+// a slice of OutputRequests by their outBailmentIDHash.
+type byOutBailmentID []OutputRequest
+
+func (s byOutBailmentID) Len() int      { return len(s) }
+func (s byOutBailmentID) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s byOutBailmentID) Less(i, j int) bool {
+	return bytes.Compare(s[i].outBailmentIDHash(), s[j].outBailmentIDHash()) < 0
+}
+
+func (s outputStatus) String() string {
+	strings := map[outputStatus]string{
+		statusSuccess: "success",
+		statusPartial: "partial-",
+		statusSplit:   "split",
+	}
+	return strings[s]
+}
+
+// Outputs returns a map of outbailment IDs to WithdrawalOutputs for all outputs
+// requested in this withdrawal.
+func (s *WithdrawalStatus) Outputs() map[OutBailmentID]*WithdrawalOutput {
+	return s.outputs
+}
+
+// Sigs returns a map of ntxids to signature lists for every input in the tx
+// with that ntxid.
+func (s *WithdrawalStatus) Sigs() map[Ntxid]TxSigs {
+	return s.sigs
+}
+
+// Fees returns the total amount of network fees included in all transactions
+// generated as part of a withdrawal.
+func (s *WithdrawalStatus) Fees() btcutil.Amount {
+	return s.fees
+}
+
+// NextInputAddr returns the votingpool address that should be used as the
+// startAddress of subsequent withdrawals.
+func (s *WithdrawalStatus) NextInputAddr() WithdrawalAddress {
```

Comment:
Shouldn't this function be calculating/setting the next one as well so subsequent calls will continue getting the next one?